ui/pip_button.js

```diff
@@ -106,117 +105,23 @@ shaka.ui.PipButton = class extends shaka.ui.Element {
     this.eventManager.listen(this.player, 'trackschanged', () => {
       this.onTracksChanged_();
     });
-  }
-
-
-  /**
-   * @return {boolean}
-   * @private
-   */
-  isPipAllowed_() {
-    return ('documentPictureInPicture' in window) ||
-      (document.pictureInPictureEnabled && !this.video.disablePictureInPicture);
-  }
-
 
-  /**
-   * @return {!Promise}
-   * @private
-   */
-  async onPipClick_() {
-    try {
-      if ('documentPictureInPicture' in window) {
-        await this.toggleDocumentPictureInPicture_();
-        return;
-      }
-      if (!document.pictureInPictureElement) {
-        // If you were fullscreen, leave fullscreen first.
-        if (document.fullscreenElement) {
-          document.exitFullscreen();
-        }
-        await this.video.requestPictureInPicture();
-      } else {
-        await document.exitPictureInPicture();
-      }
-    } catch (error) {
-      this.controls.dispatchEvent(new shaka.util.FakeEvent(
-          'error', (new Map()).set('detail', error)));
-    }
-  }
-
-  /**
-   * The Document Picture-in-Picture API makes it possible to open an
-   * always-on-top window that can be populated with arbitrary HTML content.
-   * https://developer.chrome.com/docs/web-platform/document-picture-in-picture
-   * @private
-   */
-  async toggleDocumentPictureInPicture_() {
-    // Close Picture-in-Picture window if any.
-    if (window.documentPictureInPicture.window) {
-      window.documentPictureInPicture.window.close();
-      this.onLeavePictureInPicture_();
-      return;
-    }
-
-    // Open a Picture-in-Picture window.
-    const pipPlayer = this.videoContainer_;
-    const rectPipPlayer = pipPlayer.getBoundingClientRect();
-    const pipWindow = await window.documentPictureInPicture.requestWindow({
-      width: rectPipPlayer.width,
-      height: rectPipPlayer.height,
-    });
-
-    // Copy style sheets to the Picture-in-Picture window.
-    this.copyStyleSheetsToWindow_(pipWindow);
-
-    // Add placeholder for the player.
-    const parentPlayer = pipPlayer.parentNode || document.body;
-    const placeholder = this.videoContainer_.cloneNode(true);
-    placeholder.style.visibility = 'hidden';
-    placeholder.style.height = getComputedStyle(pipPlayer).height;
-    parentPlayer.appendChild(placeholder);
-
-    // Make sure player fits in the Picture-in-Picture window.
-    const styles = document.createElement('style');
-    styles.append(`[data-shaka-player-container] {
-      width: 100% !important; max-height: 100%}`);
-    pipWindow.document.head.append(styles);
-
-    // Move player to the Picture-in-Picture window.
-    pipWindow.document.body.append(pipPlayer);
-    this.onEnterPictureInPicture_();
-
-    // Listen for the PiP closing event to move the player back.
-    this.eventManager.listenOnce(pipWindow, 'pagehide', () => {
-      placeholder.replaceWith(/** @type {!Node} */(pipPlayer));
+    this.eventManager.listen(this.player, 'trackschanged', () => {
```

Comment:
The diff makes it look like you have duplicated this listener.  Am I reading that right?  If so, can you please fix it?

Reply:
Sorry, changed!